nil pointer dereference in kubectl/plugins/loader.go:70

New kubectl plugin loader introduced by https://github.com/kubernetes/kubernetes/pull/37499 does not check `walkErr` before deref'ing `fileInfo`.  This occurs if the plugins directory does not exist or the user running `kubectl` doesn't have permission on the plugin directory.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/plugins/loader.go#L70

@derekwaynecarr @fabianofranz 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x19fbf06]

goroutine 1 [running]:
k8s.io/kubernetes/pkg/kubectl/plugins.(*DirectoryPluginLoader).Load.func1(0xc420cf2150, 0x2b, 0x0, 0x0, 0x2ba3020, 0xc420cf21b0, 0x0, 0x0)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/plugins/loader.go:70 +0x46
path/filepath.walk(0xc420cf2090, 0x22, 0x2bbc940, 0xc420cee410, 0xc42030e140, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:372 +0x2fe
path/filepath.walk(0xc42028a150, 0xe, 0x2bbc940, 0xc420958000, 0xc42030e140, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:376 +0x414
path/filepath.walk(0x1ed8c58, 0xa, 0x2bbc940, 0xc42083dba0, 0xc42030e140, 0x0, 0x10)
	/usr/local/go/src/path/filepath/path.go:376 +0x414
path/filepath.Walk(0x1ed8c58, 0xa, 0xc42030e140, 0xa, 0x0)
	/usr/local/go/src/path/filepath/path.go:398 +0x14c
k8s.io/kubernetes/pkg/kubectl/plugins.(*DirectoryPluginLoader).Load(0xc42030e030, 0x2cd3e20, 0x0, 0x0, 0x2cd3e20, 0x0)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/plugins/loader.go:98 +0x171
k8s.io/kubernetes/pkg/kubectl/plugins.TolerantMultiPluginLoader.Load(0xc42000a480, 0x2, 0x2, 0x40f448, 0x0, 0x1c00d60, 0xc42000a701, 0x2cd3e20)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/plugins/loader.go:179 +0x93
k8s.io/kubernetes/pkg/kubectl/plugins.(*TolerantMultiPluginLoader).Load(0xc42000a640, 0x2cd3e20, 0x40cc18, 0x1c98f20, 0xc42000a720, 0xc420887b48)
	<autogenerated>:5 +0x5f
k8s.io/kubernetes/pkg/kubectl/plugins.TolerantMultiPluginLoader.Load(0xc42000a680, 0x2, 0x2, 0xc420887b88, 0x1a1e3ba, 0xc420435460, 0x2ba3d60, 0xc42000a720)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/plugins/loader.go:179 +0x93
k8s.io/kubernetes/pkg/kubectl/plugins.(*TolerantMultiPluginLoader).Load(0xc42000a720, 0x2ba3d60, 0xc42000a720, 0xc4202f9200, 0x1f759b0, 0xc)
	<autogenerated>:5 +0x5f
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdPlugin(0x2bc7180, 0xc42024d080, 0x2ba2f60, 0xc42000c010, 0x2ba2fa0, 0xc42000c018, 0x2ba2fa0, 0xc42000c020, 0xc4208a0540)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/plugin.go:44 +0x60
k8s.io/kubernetes/pkg/kubectl/cmd.NewKubectlCommand(0x2bc7180, 0xc42024d080, 0x2ba2f60, 0xc42000c010, 0x2ba2fa0, 0xc42000c018, 0x2ba2fa0, 0xc42000c020, 0x1b0912e)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go:371 +0x16f3
k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:38 +0xc7
main.main()
	/home/sjennings/projects/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:26 +0x22
```

--- a/pkg/kubectl/plugins/loader.go
+++ b/pkg/kubectl/plugins/loader.go
@@ -67,7 +67,7 @@ func (l *DirectoryPluginLoader) Load() (Plugins, error) {
 	// read the base directory tree searching for plugin descriptors
 	// fails silently (descriptors unable to be read or unmarshalled are logged but skipped)
 	err = filepath.Walk(base, func(path string, fileInfo os.FileInfo, walkErr error) error {
-		if fileInfo.IsDir() || fileInfo.Name() != PluginDescriptorFilename || walkErr != nil {
+		if walkErr != nil || fileInfo.IsDir() || fileInfo.Name() != PluginDescriptorFilename {
 			return nil
 		}
 

